src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs

```diff
@@ -509,6 +515,55 @@ public Task<Block> GetAsync(uint256 hash)
             return task;
         }
 
+        /// <summary>
+        /// Get multiple blocks from the database by their block hashes.
+        /// </summary>
+        /// <param name="hashes">The block hashes.</param>
+        public Task<List<Block>> GetBlocksAsync(List<uint256> hashes)
+        {
+            this.logger.LogTrace("({0}:'{1}')", nameof(hashes.Count), hashes.Count);
+            Guard.NotNull(hashes, nameof(hashes));
+
+            var results = new Dictionary<uint256, Block>();
+
+            Task<List<Block>> task = Task.Run(() =>
+            {
+                this.logger.LogTrace("()");
+
+                using (DBreeze.Transactions.Transaction transaction = this.DBreeze.GetTransaction())
+                {
+                    transaction.ValuesLazyLoadingIsOn = false;
+
+                    // Access hashes in sorted order.
+                    var byteListComparer = new ByteListComparer();
+                    hashes.Sort((pair1, pair2) => byteListComparer.Compare(pair1.ToBytes(), pair2.ToBytes()));
+
+                    foreach (var hash in hashes)
+                    {
+                        byte[] key = hash.ToBytes();
+                        Row<byte[], Block> blockRow = transaction.Select<byte[], Block>("Block", key);
+                        if (blockRow.Exists)
+                        {
+                            results[hash] = blockRow.Value;
```

Comment:
we probably want to log every such success (and/or failure) to retrieve a block

Reply:
Since we have no logging on the input of the method in terms of the full list, we should log both the success and failures